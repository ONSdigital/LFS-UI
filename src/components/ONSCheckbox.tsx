import React, { Component, ChangeEvent } from 'react';

interface Props{
  label?: string,
  id?: string,
  onChange : ((...props: any[]) => void) | undefined,
  checked? : boolean
}

interface State{
  value: boolean
}

export class ONSCheckbox extends Component <Props, State>{
  value: boolean = this.props.checked === true ? true : false;
  constructor(props : Props) {
    super(props);
    this.state = {value: props.checked === true ? true : false};
  }

  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    if(this.props.onChange !== undefined){
      this.props.onChange(e);
    }
    this.value = e.target.checked;
    this.setState({value: e.target.checked})
    
  }

  componentWillReceiveProps(newProps: Props){
    if(newProps.checked !== undefined){
      this.setState({value: newProps.checked})
    }
  }

    render() {
    if(this.props.label !== undefined){
        return (
            <span className="checkbox ">
                <input
                    type="checkbox"
                    id={this.props.id}
                    checked={this.state.value}
                    onChange={(e) => this.handleChange(e)}
                    className="checkbox__input js-checkbox "
                    value="bacon"
                />
                <label id="bacon-label" className="checkbox__label">
                    {this.props.label}
                </label>
            </span>
        );
    }else{

    }
    return (
        <span style={{width:"unset"}} className="checkbox">
            <input
                type="checkbox"
                id={this.props.id}
                className="checkbox__input js-checkbox "
                checked={this.state.value}
                onChange={(e) => this.handleChange(e)}
                style={{position:"unset"}}
            />
        </span> 
    );
  }
}
