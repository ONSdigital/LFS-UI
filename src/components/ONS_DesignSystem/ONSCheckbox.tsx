import React, {CSSProperties} from 'react';

interface Props {
    label?: string,
    id: string,
    onChange?: ((...props: any[]) => void) | undefined,
    checked?: boolean,
    style?: CSSProperties
    disabled?: boolean
}

export const ONSCheckbox = (props: Props) => (
    props.label !== undefined ?
        <p className="checkboxes__item" style={props.style}>
            <span className="checkbox ">
                <input
                    disabled={props.disabled}
                    type="checkbox"
                    id={props.id}
                    checked={props.checked}
                    onChange={(e) => props.onChange !== undefined && props.onChange(e)}
                    className="checkbox__input js-checkbox "
                    value={props.id}
                    name={props.label}
                />
                <label className="checkbox__label " htmlFor={props.id}>
                    {props.label}
                </label>
            </span>
        </p>
        :
        <span style={{width: "unset"}} className="checkbox">
            <input
                disabled={props.disabled}
                type="checkbox"
                id={props.id}
                className="checkbox__input js-checkbox "
                checked={props.checked}
                onChange={(e) => props.onChange && props.onChange(e)}
                style={{position: "unset"}}
            />
        </span>
);
