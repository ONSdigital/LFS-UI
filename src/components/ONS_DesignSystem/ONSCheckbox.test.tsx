import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {ONSCheckbox} from "./ONSCheckbox";
import {Builder, By, WebDriver} from 'selenium-webdriver';

describe("ONS Checkbox Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    const checkboxProps = {
        id: '12',
        onCheckboxClick: jest.fn(),
    };

    const checkboxWithLabelProps = {
        label: "Enable",
        id: '12',
        onCheckboxClick: jest.fn(),
        checked: true
    };

    const checkboxCheckedProps = {
        label: "Enable",
        id: '12',
        checked: true,
    };

    function wrapper(render: any, props: any) {
        return render(
            <ONSCheckbox id={props.id}
                         label={props.label}
                         onChange={props.onCheckboxClick}
                         disabled={props.disabled}
                         checked={props.checked}
                         style={props.style}/>)
    }
    
    let url = 'http://localhost:3000';
    let browser: WebDriver;

    // beforeEach(async  () => browser = await new Builder().forBrowser('chrome').build())

    // afterEach(async () => {
    //     if (browser !== undefined) {
    //         await browser.quit();
    //     }
    // });

    it("should render correctly", () => expect(wrapper(shallow, checkboxProps).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, checkboxWithLabelProps).find("ONSCheckbox").getElement().props.label).toEqual(checkboxWithLabelProps.label);
    });

    it("should render without label if prop not passed in", () => {
            expect(wrapper(mount, checkboxProps).find("ONSCheckbox").getElement().props.label).toBeUndefined();
        }
    );

    it("should render with the correct checked status", () => {
        expect(wrapper(mount, checkboxCheckedProps).find("ONSCheckbox").getElement().props.checked).toEqual(true);
    });

    it('simulates change events', () => {
        wrapper(mount, checkboxProps).find('input').simulate('change')
        expect(checkboxProps.onCheckboxClick).toHaveBeenCalled()

    });

    

    // it('matches snapshot', () => {
    //     expect(wrapper(mount, checkboxProps)).toMatchSnapshot()
    // });
});