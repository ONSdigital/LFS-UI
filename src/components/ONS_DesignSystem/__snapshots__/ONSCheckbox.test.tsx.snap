// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Checkbox Test matches Snapshot 1`] = `ShallowWrapper {}`;
